src/components/DropdownMenu/DropdownMenu.tsx

```diff
@@ -45,107 +60,78 @@ export const DropdownMenu: React.FC<Props> = ({
-    let activeItem = null;
-
-    childRefs.current.map((item: HTMLLIElement) => {
-      if (item.querySelector(':first-child') === document.activeElement) {
-        activeItem = item;
+    if (e.key === R_ARROW_KEYCODE || e.key === D_ARROW_KEYCODE) {
+      if (focusIndex < refs.current.list.length - 1) {
+        focusIndex++;
+      } else {
+        focusIndex = 0;
       }
-      return item;
-    });
-
-    if (!activeItem) return;
-
-    const index = childRefs.current.indexOf(activeItem); /* 2 */
-    const next = index === childRefs.current.length - 1 ? 0 : index + 1; /* 2 */
-
-    const prev = index === 0 ? childRefs.current.length - 1 : index - 1; /* 2 */
-
-    if ([R_ARROW_KEYCODE, D_ARROW_KEYCODE].includes(e.key)) {
-      /* 3 */
-      childRefs.current[next]
-        ?.querySelector<HTMLButtonElement | HTMLAnchorElement>(':first-child')
-        ?.focus();
-    } else if ([L_ARROW_KEYCODE, U_ARROW_KEYCODE].includes(e.key)) {
-      /* 4 */
-      childRefs.current[prev]
-        ?.querySelector<HTMLButtonElement | HTMLAnchorElement>(':first-child')
-        ?.focus();
+      focusItem(focusIndex);
+      // prevents page from scrolling
+      e.preventDefault();
     }
-  };
-
-  /**
-   * Pass props down to children
-   * 1) Using type 'any' here because TS requires typechecking at this point, but these elements have already been typechecked
-   * 2) Cycle through children and pass down ref
-   */
-  const childrenWithProps = React.Children.map(
-    children,
-    (child: ReactNode, i: number) => {
-      // Checking isValidElement is the safe way and avoids a typescript
-      // error too.
-      if (React.isValidElement(child)) {
-        if (isReactFragment(child)) {
-          const newChildren = child.props.children;
-          return newChildren.map((item: ReactNode, i: number) => {
-            return React.cloneElement<Props>(
-              item as any /* 1 */,
-              {
-                ref: (el: HTMLLIElement) => (childRefs.current[i] = el) /* 2 */,
-              } as any /* 1 */,
-            );
-          });
-        } else {
-          return React.cloneElement<Props>(
-            child as any /* 1 */,
-            {
-              ref: (el: HTMLLIElement) => (childRefs.current[i] = el) /* 2 */,
-            } as any /* 1 */,
-          );
-        }
+    if (e.key === L_ARROW_KEYCODE || e.key === U_ARROW_KEYCODE) {
+      if (focusIndex > 0) {
+        focusIndex--;
+      } else {
+        focusIndex = refs.current.list.length - 1;
       }
-    },
-  );
+      focusItem(focusIndex);
+      // prevents page from scrolling
+      e.preventDefault();
+    }
+    if (e.key === HOME_KEYCODE) {
+      focusIndex = 0;
+      focusItem(focusIndex);
+      // prevents page from scrolling
+      e.preventDefault();
+    }
+    if (e.key === END_KEYCODE) {
+      focusIndex = refs.current.list.length - 1;
+      focusItem(focusIndex);
+      // prevents page from scrolling
+      e.preventDefault();
+    }
```

Comment:
Oo nice

Reply:
Yea was reading https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/menu_role#keyboard_interactions and realized we were missing this functionality (think Headless menu also handles this but that's out of scope for now)